samples/graph-activity-feed-broadcast/csharp/ActivityFeedBroadcast/Controllers/HomeController.cs

```diff
@@ -0,0 +1,229 @@
+            _httpClientFactory = httpClientFactory;
+            _httpContextAccessor = httpContextAccessor;
+            _taskList = taskList;
+        }
+
+        // Get broadcast page.
+        [Route("broadcast")]
+        public ActionResult Broadcast()
+        {
+            return View("Index");
+        }
+
+        // Get broadcast message Id.
+        [HttpGet]
+        [Route("MessageDetails")]
+        public ActionResult GetRequestByID(string taskId)
+        {
+            var currentTaskList = new List<BroadcastInfo>();
+            _taskList.TryGetValue("taskList", out currentTaskList);
+
+            if (currentTaskList == null)
+            {
+                ViewBag.Message = "No record found";
+            }
+            else
+            {
+                var request = currentTaskList.FirstOrDefault(p => p.taskId.ToString() == taskId);
+                ViewBag.TaskDetails = request;
+            }
+            return View("Message");
+        }
+
+        // Send notification toorganisation.
+        [HttpPost]
+        [Route("SendNotificationToOrganisation")]
+        public async Task<ActionResult> SendNotificationToOrganisation(BroadcastInfo taskInfo)
+        {
+            try
+            {
+                var currentTaskList = new List<BroadcastInfo>();
+                List<BroadcastInfo> taskList = new List<BroadcastInfo>();
+                _taskList.TryGetValue("taskList", out currentTaskList);
+
+                taskInfo.taskId = Guid.NewGuid();
+
+                if (currentTaskList == null)
+                {
+                    taskList.Add(taskInfo);
+                    _taskList.AddOrUpdate("taskList", taskList, (key, newValue) => taskList);
+                    ViewBag.TaskList = taskList;
+                }
+                else
+                {
+                    currentTaskList.Add(taskInfo);
+                    _taskList.AddOrUpdate("taskList", currentTaskList, (key, newValue) => currentTaskList);
+                    ViewBag.TaskList = currentTaskList;
+                }
+
+                var graphClient = SimpleGraphClient.GetGraphClient(taskInfo.access_token);
+                var graphClientApp = SimpleGraphClient.GetGraphClientforApp(_configuration["AzureAd:MicrosoftAppId"], _configuration["AzureAd:MicrosoftAppPassword"], _configuration["AzureAd:TenantId"]);
+                var usersList = await graphClient.Users
+                                 .Request()
+                                    .GetAsync();
+
+                var installedAppsForCurrentUser = await graphClient.Users[taskInfo.userId].Teamwork.InstalledApps
+                                   .Request()
+                                   .Expand("teamsAppDefinition")
+                                   .GetAsync();
+                var appId = installedAppsForCurrentUser.Where(id => id.TeamsAppDefinition.DisplayName == "Activity feed broadcast").Select(x => x.TeamsAppDefinition.TeamsAppId);
```

Comment:
We should provide this information about the Display name being used for the condition.

Reply:
added comment and added to readme